openhexa/toolbox/hexa/api.py

```diff
@@ -0,0 +1,67 @@
+from requests import Session
+import typing
+
+
+class NotFound(Exception):
+    """Errors related to an element not found."""
+
+    pass
+
+
+class OpenHEXAClient:
+    def __init__(self, base_url):
+        self.url = base_url.rstrip("/")
+        self.session = Session()
+        self.session.headers.update({"Content-Type": "application/json", "User-Agent": "OpenHEXA Python Client"})
+
+    def authenticate(
+        self,
+        with_credentials: typing.Optional[tuple[str, str]] = None,
+        with_token: typing.Optional[str] = None,
+    ):
+        """
+        with_credentials: tuple of email and password
+        with_token: JWT token
+        """
+        if with_credentials:
+            resp = self._graphql_request(
+                """
+                mutation Login($input: LoginInput!) {
+                    login(input: $input) {
+                        success
+                    }
+                }
+            """,
+                {
+                    "input": {
+                        "email": with_credentials[0],
+                        "password": with_credentials[1],
+                    }
+                },
+            )
+            resp.raise_for_status()
+            data = resp.json()["data"]
+            if data["login"]["success"]:
+                self.session.headers["Cookie"] = resp.headers["Set-Cookie"]
+            else:
+                raise Exception("Login failed")
+        elif with_token:
+            self.session.headers.update({"Authorization": f"Bearer {with_token}"})
+        try:
+            self.query("""query{me {user {id}}}""")
```

Comment:
Yes, you're right.

Reply:
@qgerome Indeed the login when 2FA is enabled the login with username/password will not work and with token we cannot query the User (`"""query{me {user {id}}}"""`) because of the check `request.user.is_verified()` (no attribute is_verified). I propose to solve it to check 
`  if has_configured_two_factor(request.user) and getattr(request, "bypass_two_factor", False):
[....]`
WDYT?

Reply:
Yes, I have arrived to same code blocks as Cheick when i was looking into why it wasn't working for Jipe and Martin at some point.

Reply:
I would not skip 2fa on backend side, as it removes the point of having one. To use the toolbox in the pipeline with the token the user with 2FA enabled, I  would suggest  to avoid executing the "query me" to check if he is authenticated
It fits better with what we offer